src/main/java/co/rsk/bitcoinj/script/Script.java

```diff
@@ -482,50 +481,46 @@ public Script getScriptSigWithSignature(Script scriptSig, byte[] sigBytes, int i
         return ScriptBuilder.updateScriptWithSignature(scriptSig, sigBytes, index, sigsPrefixCount, sigsSuffixCount);
     }
 
-    private RedeemScriptParser getRedeemScriptParser() {
-        if (redeemScriptParser == null){
-            redeemScriptParser = RedeemScriptParserFactory.get(chunks);
-        }
-        return redeemScriptParser;
-    }
-
     /**
      * Returns the index where a signature by the key should be inserted. Only applicable to
      * a P2SH scriptSig.
      */
     public int getSigInsertionIndex(Sha256Hash hash, BtcECKey signingKey) {
+        List<ScriptChunk> rawChunks = this.getChunks();
         // Iterate over existing signatures, skipping the initial OP_0, the final redeem script
         // and any placeholder OP_0 sigs.
-        List<ScriptChunk> existingChunks = chunks.subList(1, chunks.size() - 1);
-        ScriptChunk redeemScriptChunk = chunks.get(chunks.size() - 1);
-        checkNotNull(redeemScriptChunk.data);
```

Comment:
Is it safe to remove this check?

Reply:
This implementation is the one from master, see it [here](https://github.com/rsksmart/bitcoinj-thin/blob/master/src/main/java/co/rsk/bitcoinj/script/StandardRedeemScriptParser.java#L55). However, I will rollback at the one accepted in the integration branch, implemented by Juli, see it [here](https://github.com/rsksmart/bitcoinj-thin/blob/fed-refactor-integration/src/main/java/co/rsk/bitcoinj/script/Script.java#L496C16-L496C36).